### ISSUE TITLE
Does hyperscan thread-safed?

### ISSUE THREAD

**Issue by Vincent717** [intel/hyperscan#114]
Hi all, I am trying to run hyperscan on python, with multi-threading. Seems like it will throw errors like:  
`Fatal Python error: ceval: tstate mix-up`
and 
`Fatal Python error: GC object already tracked`

Does it mean that hyperscan is not thread-safed and cannot be used with multi-threading?

Thx.
Vincent

**Comment by xiangwang1** [comment]
Hi Vincent,

Hyperscan can be used for multi-threading cases. Each scan thread can share a common database but needs to have its own scratch space if matching in parallel. It'll be good if you can provide more details.

Thanks,
Xiang
